apps/src/blockly/addons/functionEditor.js

```diff
@@ -318,33 +314,6 @@ export default class FunctionEditor {
     return returnValue;
   }
 
-  /**
-   * Constructs the blocks required by the flyout for the procedure category.
-   * Modeled after core Blockly procedures flyout category, but excludes unwanted blocks.
-   * Derived from core Google Blockly:
-   * https://github.com/google/blockly/blob/5a23c84e6ef9c0b2bbd503ad9f58fa86db1232a8/core/procedures.ts#L202-L287
-   * @param {WorkspaceSvg} workspace The workspace containing procedures.
-   * @returns an array of XML block elements
-   */
-  // Equivalent to blockly-samples registerToolboxCategoryCallback->modalProceduresToolboxCallback
-  flyoutCategory(workspace, includeNewButton = true) {
-    const blockList = [];
-    if (includeNewButton) {
-      blockList.push({
-        kind: 'button',
-        text: 'Create a Function',
-        callbackKey: 'newProcedureCallback',
-      });
-    }
-
-    // Get all the procedures from the workspace and create call blocks for them
-    workspace
-      .getProcedureMap()
-      .getProcedures()
-      .forEach(procedure => blockList.push(this.createCallBlock(procedure)));
-    return blockList;
-  }
-
   createCallBlock(procedure) {
```

Comment:
I think we'll need to include the id in `extraState` added in this function to the equivalent function in `procedureBlocks`. I also didn't need to include `fields` for this to work.

Reply:
(and then we can probably remove this function?)

Reply:
The functions in `procedureBlocks` and `behaviorBlocks` don't operate on the procedures, just the procedure names, so we can't get the ID off them yet like we do here, right? Do we need to modify Mike's functions to use the new model? I'm currently testing whether modifying the code that searches the workspaces to return the name and id is sufficient.
```
allFunctionNames.sort().forEach(name => {
    blockList.push({
      kind: 'block',
      type: 'procedures_callnoreturn',
      extraState: {
        name: name,
      },
      fields: {
        NAME: name,
      },
    });
  });
```

Reply:
I'm not sure, this configuration is a bit of a black box for me. If it works as expected it might be fine, but I'm pretty sure we needed `id` in order to make things work.